src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -168,6 +168,62 @@ public RTPEncodingImpl(
     }
 
     /**
+     * Applies frame boundaries heuristics to frames a and b, assuming a
+     * predates/is older than b.
+     *
+     * @param a the old {@link FrameDesc}.
+     * @param b the new {@link FrameDesc}
+     */
+    private static void applyFrameBoundsHeuristics(FrameDesc a, FrameDesc b)
+    {
+        int end = a.getEnd(), start = b.getStart();
+        if (end != -1 && start != -1)
+        {
+            // No need for heuristics.
+            return;
+        }
+
+        long tsDiff = (b.getTimestamp() - a.getTimestamp()) & 0xFFFFFFFFL;
+        if (tsDiff > (1L << 30) && tsDiff < (-(1L << 30) & 0xFFFFFFFFL))
```

Comment:
We already assume that "a" comes before "b" when we choose to take the end of "a" and the start of "b" rather than the other way around. So I don't think we should apply the same logic in the case when tsDiff > (-(1L << 30) & 0xFFFFFFFFL) (i.e. "b" comes before "a"), because we would be working with the start of the first frame and the end of the second frame.